types/danish-ssn/index.d.ts

```diff
@@ -0,0 +1,16 @@
+// Type definitions for danish-ssn 2.1.1
+// Project: https://github.com/mathiasvr/danish-ssn/
+// Definitions by: Yuriy Guzenko <https://github.com/yuriyg86>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export function isValid(ssn: string): boolean;
+export function validate(ssn: string): string;
+export function validForDate(date: Date): string[];
+
+export interface DanishSsnStatus {
+    cpr: string;
+    valid: boolean;
+    date: Date;
+    sex: 'Female' | 'Male';
+}
+export default function(ssn: string): DanishSsnStatus;
```

Comment:
Use the commonjs syntax

```ts
function danishSsn(ssn: string): DanishSsnStatus;
export = danishSsn;
```

Consumers of the package who want to use `import ssn from 'danish-ssn'` will need to add `"esModuleInterop": true`, but Definitely Typed tests need to be backward compatible to show that the types will work with old code bases.

Reply:
Thank you for review. I already tried your solution. But I don't know how to export other 3 functions in this case. If I do:

```
export function isValid(ssn: string): boolean;
export function validate(ssn: string): string;
export function validForDate(date: Date): string[];

export interface DanishSsnStatus {
    cpr: string;
    valid: boolean;
    date: Date;
    sex: 'Female' | 'Male';
}
declare function cpr(ssn: string): DanishSsnStatus;
export = cpr;
```

I am getting error:

```
ERROR: 17:1  expect  TypeScript@4.6 compile error: 
An export assignment cannot be used in a module with other exported elements.
```

Reply:
You have to use declaration merging, which is a fairly obscure feature that comes up on DT quite a bit:

```ts
declare namespace cpr {
  export function isValid(...
  // other functions and interfaces go here...
}
declare function cpr(ssn: string): DanishSsnStatus;
```

The test's import statement will still work, but you'll need a separate one to import the commonjs 'default': `import ssn = require('danish-ssn')`.

Reply:
Thanks. Works for me. But had to add `"strict-export-declare-modifiers": false` to tslint to pass tests

Reply:
remove `exports` from within namespace, as in other comment

Reply:
Done. Huge thanks peterblazejewicz